src/common/system_signal_handling.c

```diff
@@ -178,7 +216,33 @@ void dt_set_signal_handlers()
     fprintf(stderr, "[dt_set_signal_handlers] error: signal(SIGSEGV) returned SIG_ERR: %i (%s)\n", errsv,
             strerror(errsv));
   }
-#endif
+#else
+/*
+Set up exception handler for backtrace on Windows
+Works when there is NO SIGSEGV handler installed
+
+Must be set up _after_: InitializeMagick(),
+as GraphicsMagick is overwriting SetUnhandledExceptionFilter
+and all other signals in its init function.
+The first invokation of the dt_set_signal_handlers() is _before_ InitializeMagick(),
+therefore no point to set up exception handler
+It should be set up only at the second invocation of  dt_set_signal_handlers(),
```

Comment:
[00:24:53] <LebedevRI> can you call SetUnhandledExceptionFilter() from src/common/system_signal_handling.c?
[00:24:56] <peterbud> saving it in the first place and then restore?
[00:24:59] <LebedevRI> yeah
[00:25:19] <peterbud> whoo, indeed good idea!
[00:29:57] <LebedevRI> i think it should be like this  0. call dt_set_unhandled_exception_handler_win(), ALWAYS  1. if(1 == _times_handlers_were_set) call SetUnhandledExceptionFilter(), and save the result  2. immediately call SetUnhandledExceptionFilter() with the saved result
[00:30:15] <LebedevRI> assuming you can call SetUnhandledExceptionFilter(), that should work?